cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/ReplicationStatusKey.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.replication;
+
+import co.cask.cdap.api.common.Bytes;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+
+/**
+ * Handling of Row keys for the Replication State Table
+ */
+public final class ReplicationStatusKey {
+
+  private final byte[] key;
+
+  public ReplicationStatusKey(byte[] prefix, byte[] regionName, byte[] regionServerID) {
+    byte[] rowKey;
+    rowKey = Bytes.add(Bytes.toBytes(prefix.length), prefix);
+    rowKey = Bytes.add(rowKey, Bytes.toBytes(regionName.length), regionName);
+    rowKey = Bytes.add(rowKey, Bytes.toBytes(regionServerID.length), regionServerID);
+    this.key = rowKey;
+  }
+
+  public ReplicationStatusKey(byte[] key) {
+    this.key = key;
+  }
+
+  /**
+   * @return the underlying key
+   */
+  public byte[] getKey() {
+    return key;
+  }
+
+  /**
+   * Splits the keys into a {@link Splitter} which exposes the parts that comprise the key.
+   */
+  public Splitter split() {
```

Comment:
The user still needs to know how the key is constructed because these functions can only be called in this sequence. Thats why may be its better to call it getBytes and leave the user to interpret based on the given order. MDSKey.java is another example in code.